### PR TITLE
Add storage FFI calls

### PR DESCRIPTION
**Pull request by Lipt0nas**

### CODE REVIEW

**Review by LukasPukenis — CHANGES_REQUESTED** [review]
That's very nice. What is missing - Python wrappers for the new APIs and also at least one test case for each would be nice. Several for retrieval would be even nicer :)

File: norddrop/src/device/mod.rs
Comment (matszczygiel) [reply norddrop/src/device/mod.rs]: I would go even further `transfers_since` https://rust-lang.github.io/api-guidelines/naming.html#c-getter

Comment (Lipt0nas) [reply norddrop/src/device/mod.rs]: fair enough, but I guess we should keep the binding-side naming with the `get` prefix

File: norddrop/src/device/mod.rs
Comment (matszczygiel) [inline norddrop/src/device/mod.rs:288]: These are wrappers for wrappers. I propose keeping the `Storage` field in the `instance` and calling it directly. It will be less code and less mental overhead IMHO.

File: norddrop/src/device/mod.rs
Comment (matszczygiel) [reply norddrop/src/device/mod.rs]: yeah, that's true ;)

File: norddrop/src/device/mod.rs
Comment (Lipt0nas) [reply norddrop/src/device/mod.rs:288]: You mean adding a reference to `StorageDispatch` in `NorddropFFI` to skip going trough `Service` for the calls?

File: norddrop/src/ffi/mod.rs
Comment (matszczygiel) [inline norddrop/src/ffi/mod.rs:317]: I'm unsure if returning `*mut c_char` is good here. With this in the `new_transfer` function, we already shoot ourselves in the foot. Because getting `null` can mean anything. 
Maybe we can do something like `transfers_since(dev: &norddrop, since: i64, out_transfers: **mut c_char) -> norddrop_result` and write the `char *` to the `out_transfers` on success? I'm not sure how swig will handle that TBH

Comment (LukasPukenis) [reply norddrop/src/ffi/mod.rs:317]: I think this is a double-edged sword. The API would be better but I smell binding issues and problems by accepting pointer to pointer. This is fine to be left as tech debt for future revisions when we already have features going

Comment (matszczygiel) [reply norddrop/src/ffi/mod.rs:317]: Yeah, it's probably true. But since it's potentially breaking change in the future I would at least try to see if Swig is a problem here.
If yes, then it's better to abandon the idea

Comment (Lipt0nas) [reply norddrop/src/ffi/mod.rs:317]: I'll have a look if it's a trivial change or not :+1:

Comment (LukasPukenis) [reply norddrop/src/ffi/mod.rs:317]: I think it's hard to quickly say if it's a good idea as Golang bindings have shown after some time that it causes trouble. Or C#. I suggest not blocking this PR due to this however exploring this idea for some small amount of time after it's been landed :)

Comment (Lipt0nas) [reply norddrop/src/ffi/mod.rs:317]: I'll create a ticket for this as a possible improvement then, to keep it in the back of our minds

File: drop-transfer/src/service.rs
Comment (matszczygiel) [inline drop-transfer/src/service.rs:146]: Minor: to reduce wrapping you can just provide getter `pub fn storage(&self) -> &Storage` and delegate to it down the line

**Review by matszczygiel — APPROVED** [review]

File: drop-transfer/src/service.rs
Comment (Lipt0nas) [reply drop-transfer/src/service.rs]: Good catch on the `mut`, it remained from when it originally was an `Arc<Mutex<...>>`, As for the layers, i think id like to keep the ffi layer function clean, and leave the processing in `Service`

**Review by matszczygiel — APPROVED** [review]

**Pull request merged**
